### ISSUE TITLE
[Flaky-test] UpsertTableSegmentPreloadIntegrationTest.testSegmentAssignment()

### ISSUE THREAD

**Issue by Jackie-Jiang** [apache/pinot#11359]
```
Error:    UpsertTableSegmentPreloadIntegrationTest.testSegmentAssignment:200->waitForSnapshotCreation:216 Failed to meet condition in 120000ms, error message: Error verifying force commit operation on table!
```

https://github.com/apache/pinot/actions/runs/5875246234/job/15932781085

**Comment by KKcorps** [comment]
Ack. I think it might be cause I am not enforcing during ingestion that all segments should have at least one primary keys while enforcing that all segments have a valid snapshot file. Will take a look

**Comment by ege-st** [comment]
@KKcorps can this ticket be closed?

**Comment by Jackie-Jiang** [comment]
Fixed in #11458
